### PR TITLE
Auto-detect the language in the gradle plugin

### PR DESCRIPTION
**Pull request by eshepelyuk**
* detect languages files and apply java or scala plugin

References #716 

@raboof @ignasi35 plz take a look

### CODE REVIEW

File: gradle-plugin/src/test/groovy/helper/BaseSpec.groovy [hunk 1]
```diff
@@ -39,7 +41,11 @@ repositories { jcenter() }
     }
 
     AkkaGrpcPluginExtension sampleSetup(def plugin = "scala", String scala = "2.12") {
-        project.pluginManager.apply plugin
+        if (plugin == "scala" || plugin == ScalaWrapperPlugin) {
+            def scalaDir = projectDir.newFolder('src', 'main', 'scala')
+            new File(scalaDir, "test.scala").text = "object AkkaGrpc"
```
Comment (raboof) [inline gradle-plugin/src/test/groovy/helper/BaseSpec.groovy:46]: Interesting! I guess it'd be nice to flesh this out further (perhaps somehow include the quickstart example?), but that's something for the future

File: gradle-plugin/src/main/groovy/akka/grpc/gradle/AkkaGrpcPluginExtension.groovy [hunk 2]
```diff
@@ -16,21 +16,26 @@ class AkkaGrpcPluginExtension {
...
     boolean generateClient = true
     boolean generateServer = true
     boolean generatePlay = false
     boolean serverPowerApis = false
     boolean usePlayActions = false
     List<String> extraGenerators = []
 
+    private final Project project
+
+    final boolean scala
```
Comment (raboof) [inline gradle-plugin/src/main/groovy/akka/grpc/gradle/AkkaGrpcPluginExtension.groovy:28]: Ah, so this is not a plugin option but just an internal variable, and as soon as 1 .scala source file appears the project switches to 'scala' mode. I guess that's OK for most common cases.

File: gradle-plugin/src/test/groovy/helper/BaseSpec.groovy (hunk 1 above)
Comment (eshepelyuk) [reply gradle-plugin/src/test/groovy/helper/BaseSpec.groovy:46]: Excuse me , I didn't understand your intentions about this.

Comment (raboof) [reply gradle-plugin/src/test/groovy/helper/BaseSpec.groovy:46]: Sorry, I misread this fragment - nothing needs to change here for this PR ;).

File: plugin-tester-java/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-bin.zip
```
Comment (raboof) [inline plugin-tester-java/gradle/wrapper/gradle-wrapper.properties:6]: should we update https://github.com/akka/akka-grpc/pull/1054/files#diff-ee344aab74dcfb1a70eb10f7d1f113bbL23 as well?

Comment (eshepelyuk) [reply plugin-tester-java/gradle/wrapper/gradle-wrapper.properties:6]: Hmm the links goes to this same page .

Comment (raboof) [reply plugin-tester-java/gradle/wrapper/gradle-wrapper.properties:6]: I meant AkkaGrpcPlugin.groovy line ~24

Comment (eshepelyuk) [reply plugin-tester-java/gradle/wrapper/gradle-wrapper.properties:6]: I don't think so, there's no need to force users to update their Gradle. As soon there's no breaking changes  - we can easily support 5.6, while using 6.4.1 for building plugin itself. `func.GradleCompatibilitySpec` will ensure that 5.6 is still supported.

Comment (raboof) [reply plugin-tester-java/gradle/wrapper/gradle-wrapper.properties:6]: OK, great!

**Comment by raboof** [comment]
PLS BUILD

**Pull request merged**